src/source/RazorEngine.Core/Compilation/CompilerServiceBase.cs

```diff
@@ -438,6 +412,34 @@ protected virtual void Inspect(CodeCompileUnit unit)
                 inspector.Inspect(unit, ns, type, executeMethod);
         }
 #endif
+
+
+        /// <summary>
+        /// Cleans up the <see cref="CompilerServiceBase"/> instance.
+        /// </summary>
+        ~CompilerServiceBase()
```

Comment:
I don't think we need a finalizer here, since we don't have any unmanaged stuff ourself.